src/main.rs

```diff
@@ -96,6 +105,41 @@ async fn main() -> std::io::Result<()> {
         .new_game(GameOptions {}, Some("test".to_string()))
         .unwrap();
 
+    let reaper_manager_ref = game_manager_wrapper.clone();
+
+    // Game Reaper thread
+    thread::spawn(move || loop {
+        thread::sleep(Duration::from_secs(REAP_INTERVAL));
+        let test_handle = GameHandle("test".to_string());
+        let num_games = reaper_manager_ref
+            .game_manager
+            .read()
+            .unwrap()
+            .games
+            .len()
+            .clone();
+
+        let mut game_manager_mut = reaper_manager_ref.game_manager.write().unwrap();
+        let games = game_manager_mut.games.clone();
+        let mut unreaped_games: HashMap<GameHandle, std::sync::Arc<RwLock<GameWrapper>>> =
+            HashMap::new();
+        for (handle, game_wrapper) in games {
+            let creation_time = game_wrapper.read().unwrap().game.game_created;
+            if handle == test_handle || (creation_time + REAP_DURATION) > get_current_time_secs() {
+                unreaped_games.insert(handle, game_wrapper);
+            }
+        }
+        let num_reaped_games = num_games - unreaped_games.len();
+        info!(
+            "Reaper: num games: {}, num games reaped this cycle: {}",
+            num_games, num_reaped_games,
+        );
+        // Only update if the number of games changed (ie. some were reaped)
+        if unreaped_games.len() != num_games {
+            game_manager_mut.games = unreaped_games;
+        }
```

Comment:
Probs cheaper just to assign it, it's just a pointer.

Reply:
will check this. i think like, i added this when i was getting panics when I had the logic for reaping games backwards